util/src/main/java/io/kubernetes/client/util/ConfigBuilder.java

```diff
@@ -0,0 +1,272 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+package io.kubernetes.client.util;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+
+import javax.net.ssl.KeyManager;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.log4j.Logger;
+
+import okio.ByteString;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.util.Config;
+import io.kubernetes.client.util.KubeConfig;
+
+public class ConfigBuilder {
+
+	private boolean clusterMode = false;
+	private boolean defaultKubeConfigMode = false;
+	private boolean defaultClientMode = false;
+	private boolean verifyingSsl = false;
+	private String basePath = null;
+	private File certificateAuthorityFile = null;
+	private String certificateAuthorityData = null;
+	private String apiKey = null;
+	private String userName = null;
+	private String password = null;
+	private KeyManager[] keyMgrs = null;
+	private String accessToken = null;
+	private String apiKeyPrefix = null;
+	private KubeConfig kubeConfig = null;
+
+	private static final Logger log = Logger.getLogger(Config.class);
+
+	public String getUserName() {
+		return userName;
+	}
+
+	public ConfigBuilder setUserName(String userName) {
+		this.userName = userName;
+		return this;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+
+	public ConfigBuilder setPassword(String password) {
+		this.password = password;
+		return this;
+	}
+
+	public String getApiKey() {
+		return apiKey;
+	}
+
+	public ConfigBuilder setApiKey(String apiKey) {
+		this.apiKey = apiKey;
+		return this;
+	}
+
+	public String getBasePath() {
+		return basePath;
+	}
+
+	public ConfigBuilder setBasePath(String basePath) {
+		this.basePath = basePath;
+		return this;
+	}
+
+	public File getCertificateAuthorityFile() {
+		return certificateAuthorityFile;
+	}
+
+	public ConfigBuilder setCertificateAuthority(File certificateAuthorityFile) {
+		this.certificateAuthorityFile = certificateAuthorityFile;
+		this.verifyingSsl = true;
+		return this;
+	}
+
+	public String getCertificateAuthorityData() {
+		return certificateAuthorityData;
+	}
+
+	public ConfigBuilder setCertificateAuthority(String certificateAuthorityData) {
+		this.certificateAuthorityData = certificateAuthorityData;
+		this.verifyingSsl = true;
+		return this;
+	}
+
+	public ConfigBuilder setClusterMode() {
+		this.clusterMode = true;
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(KubeConfig config) {
+		this.kubeConfig = config;
+		return this;
+	}
+
+	public ConfigBuilder setDefaultKubeConfigMode() {
+		this.defaultKubeConfigMode = true;
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(String fileName) throws FileNotFoundException {
```

Comment:
It's a little weird to use `File` for CertificateAuthority and `String` here, I think consistency one way or the other is probably better.